torchok/data/datasets/classification/multihead.py

```diff
@@ -0,0 +1,134 @@
+                - `type` (str): Format of data processing. Available values: multiclass, multilabel, feature_label.
+                - `num_classes` (int): Number of classes.
+                - `dtype` (str): Data type of the torch tensors related to the target.
+
+            transform: Transform to be applied on a sample. This should have the
+                interface of transforms in `albumentations`_ library.
+            augment: Optional augment to be applied on a sample.
+                This should have the interface of transforms in `albumentations`_ library.
+            input_column: column name containing paths to the images.
+            input_dtype: Data type of the torch tensors related to the image.
+            image_format: format of images that will be returned from dataset. Can be `rgb`, `bgr`, `rgba`, `gray`.
+            rgba_layout_color: color of the background during conversion from `rgba`.
+            test_mode: If True, only image without labels will be returned.
+            lazy_init: If True, for multilabel the target variable is converted to multihot when __getitem__ is called.
+                For multiclass will check the class index to fit the range when ``__getitem__`` is called.
+        .. _albumentations: https://albumentations.ai/docs/
+        """
+        super().__init__(
+            transform=transform,
+            augment=augment,
+            input_dtype=input_dtype,
+            image_format=image_format,
+            rgba_layout_color=rgba_layout_color,
+            test_mode=test_mode
+        )
+
+        self.data_folder = Path(data_folder)
+        self.input_column = input_column
+        self.annotation_path = annotation_path
+        self.lazy_init = lazy_init
+
+        if annotation_path.endswith('.csv'):
+            self.df = pd.read_csv(self.data_folder / annotation_path)
+        elif annotation_path.endswith('.pkl'):
+            self.df = pd.read_pickle(self.data_folder / annotation_path)
+        else:
+            raise ValueError('Detection dataset error. Annotation path is not in `csv` or `pkl` format')
+
+        self.heads = []
+        if not self.test_mode:
+            for target in targets:
+                name = target['name']
+                column = target['column']
+                target_type = target['type']
+                num_classes = target.get('num_classes', None)
+                target_dtype = target.get('dtype', 'long')
+
+                if target_type in range(len(self.TARGET_TYPES)):
+                    target_type = self.TARGET_TYPES[target_type]
+
+                if target_type == 'multiclass':
+                    self.heads.append((name, column, target_type, num_classes, target_dtype))
+                elif target_type == 'multilabel':
+                    self.heads.append((name, column, target_type, num_classes, target_dtype))
+                    self.csv[column] = self.csv[column].fillna('')
```

Comment:
Is it must be `self.df` now?

Reply:
How you concatenate data in `multilabel` mode? I think you need collate function, is not it?

Reply:
default collate works fine as the shape of each target is the same